app/views/miq_request/_prov_dialog_volume_fieldset.html.haml

```diff
@@ -0,0 +1,79 @@
+- url = url_for(:action => 'prov_field_changed', :id => "#{@edit[:req_id] || 'new'}") if @edit
+- options = @options ? @options : @edit[:new]
+- if keys.any? { |key| workflow.get_field(key, dialog)[:display] && workflow.get_field(key, dialog)[:display].to_s != "hide" }
+  %h3
+    = label
+- draw_fields = true
+- counter = 1
+.form-horizontal#add-volumes-form
+  - while (draw_fields)
+    #add-volume-fieldset
+      .form-group
+        %label.col-md-2.control-label{:valign => "top"}
+          = _("Volume Name *")
+        .col-md-8
+          %input.form-control{:required => :required, :value => options[:"volume_name_#{counter}"], :type => "text", :id => "volumes__volume_name_#{counter}", :name => "volumes__volume_name_#{counter}", "data-miq_observe" => {:interval => ".5", :url => url}.to_json}
+        .col-md-2
+      .form-group
+        %label.col-md-2.control-label{:valign => "top"}
+          = _("Volume Size *")
+        .col-md-8
+          %input.form-control{:required => :required, :value => options[:"volume_size_#{counter}"], :type => "text", :id => "volumes__volume_size_#{counter}", :name => "volumes__volume_size_#{counter}", "data-miq_observe" => {:interval => ".5", :url => url}.to_json}
+        .col-md-2
+      .form-group
+        %label.col-md-2.control-label{:valign => "top"}
+          = _("Delete on Terminate?")
+        .col-md-8
+          %input{:checked => options[:"volume_delete_on_terminate_#{counter}"], :type => "checkbox", :id => "volumes__volume_delete_on_terminate_#{counter}", :name => "volumes__volume_delete_on_terminate_#{counter}", "data-miq_observe" => {:interval => ".5", :url => url}.to_json}
+        .col-md-2
+      %hr
+      %div
+    - counter += 1
+    - draw_fields = !options[:"volume_name_#{counter}"].blank? && !options[:"volume_size_#{counter}"].blank?
+
+.form-horizontal
+  .form-group
+    .col-md-10
+    .col-md-2
+      %button#add-additional-volume-button
+        Add Another Volume
+
+
+:javascript
```

Comment:
@mansam - It would probably make sense to move this script tag into its own file - seeing there is a lot of logic that could potentially be reused.